storage/src/backend/registry.rs

```diff
@@ -637,18 +631,52 @@ impl RegistryReader {
             .map_err(RegistryError::Transport)
             .map(|size| size as usize)
     }
+
+    /// When invoking concurrently, only one of the handle methods will be executed first,
+    /// then subsequent handle methods will be allowed to execute concurrently.
+    ///
+    /// Nydusd uses a registry backend which generates a surge of blob requests without
+    /// auth tokens on initial startup, this caused mirror backends (e.g. dragonfly)
+    /// to process very slowly. The method implements waiting for the first blob request
+    /// to complete before making other blob requests, this ensures the first request
+    /// caches a valid registry auth token, and subsequent concurrent blob requests can
+    /// reuse the cached token.
+    fn handle<F, T>(&self, handle: &mut F) -> T
+    where
+        F: FnMut() -> T,
+    {
```

Comment:
It seems this can be replaced by:
```ruet
let mut ret = None;
self.first.call_once(|| {
    // FIXME: when handle() returns Err, we need to reset the Once instance,
    // so that the retry logic still be executed in the Once.
    ret = Some(handle());
});
ret.unwrap_or_else(handle)
```

Reply:
Good point!